### PR TITLE
Change logic to remove variable that is never read

### PR DESCRIPTION
**Pull request by jwu910**
Just saw a minor alert on [LGTM](https://lgtm.com/projects/g/emberjs/ember.js/alerts/?mode=list) and figured it would be a quick fix.

Let me know if there are additional actions that need to be taken. Thanks!

### CODE REVIEW

File: packages/@ember/-internals/utils/lib/cache.ts [hunk 1]
```diff
@@ -8,11 +8,11 @@ export default class Cache<T, V> {
   }
 
   get(key: T): V {
-    let value = this.store.get(key);
+    let value;
```
[the remark this answers is not in the document]
Comment (jwu910) [reply packages/@ember/-internals/utils/lib/cache.ts]: We can do that.

I would also change https://github.com/emberjs/ember.js/pull/17149/files#diff-8559de67243652ea464be8025157eddbR18 to remove the binding to value and return `this.set(key, this.func(key));` instead.

Then we could also just scrap line 21 `return value;` as well

**Comment by jwu910** [comment]
@btceu I submitted an update per your recommendation.

**Review by Turbo87 — APPROVED** [review]
@krisselden any objections?

**Comment by rwjblue** [comment]
Thank you!

**Pull request merged**